versioned/spi/src/main/java/org/projectnessie/versioned/Put.java

```diff
@@ -15,24 +15,35 @@
  */
 package org.projectnessie.versioned;
 
+import com.google.common.base.MoreObjects;
+import java.util.Objects;
+import java.util.function.Supplier;
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import org.immutables.value.Value;
 import org.projectnessie.model.Content;
 import org.projectnessie.model.ContentKey;
+import org.projectnessie.nessie.relocated.protobuf.ByteString;
+import org.projectnessie.versioned.store.DefaultStoreWorker;
 
 /** Setting a new value. Can optionally declare whether the prior hash must match. */
 @Value.Immutable
-public interface Put extends Operation {
+public abstract class Put implements Operation {
 
   /**
    * The value to store for this operation.
    *
    * @return the value
    */
-  Content getValue();
+  @Value.Lazy
+  public Content getValue() {
+    return getValueSupplier().get();
+  }
+
+  protected abstract Supplier<Content> getValueSupplier();
```

Comment:
Should this be annotated with `@Value.Auxiliary` ?

Reply:
We *could*, but it wouldn't change the fact that we would still need to provide a custom `equals` and `hashCode` methods, because we need `getValue()` to be included in the computation, but `getValue()` is marked `@Lazy`.

Reply:
Technically we don't need equals & hashCode for the operations in nessie-versioned-spi - if it helps to make this simpler.

Reply:
Indeed, but sadly, tests rely heavily on equality of `Operation` instances. We'd have to revisit a lot of test code... Not sure what's the least worse :-)

Reply:
yea, then let's keep it - would be "surprising" if the Put-immutable behaves differently.